server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -379,6 +379,18 @@ public boolean isPatchFrom(TransportVersion version) {
         return onOrAfter(version) && id < version.id + 100 - (version.id % 100);
     }
 
+    /**
+     * Returns {@code true} if this version is a patch version. Transport versions are generally monotoic, that is, when comparing
+     * transport versions via {@link #compareTo(VersionId)} a later version is also temporally "newer". This, however, is not the case
+     * for patch versions, as they can be introduced at any time. There may be cases where this distinction is important, in which case
+     * this method can be used to determine if a version is a patch, and therefore, may actually be temporally newer than "later" versions.
+     *
+     * @return whether this version is a patch version.
+     */
+    public boolean isPatchVersion() {
```

Comment:
Could this be defined as an (internal) utility looking at id in TransportVersionUtils? That way it can only be within tests. I think we should avoid adding new methods like this that are really only for tests because they inevitably get used in production code and it makes it harder to change.

Reply:
Ah, yes. I didn't occur to me that the `id` field was public since this is a record. I'll do that.